### ISSUE TITLE
Cannot progress on Sandy 6-1 because Halver will not set missionstatus var to 1.

### ISSUE THREAD

**Issue by helixhamin** [DarkstarProject/darkstar#3018]
**_I have:_**

- [x] searched existing issues (http://github.com/darkstarproject/darkstar/issues/) to see if the issue I am posting has already been addressed or opened by another contributor
- [x] checked the commit log to see if my issue has been resolved since my server was last updated

**_Client Version_** (type `/ver` in game) **:**
30160329_0

**_Server Version_** (type `@revision` in game) **:**
https://github.com/DarkstarProject/darkstar/commit/aa2124945c3b872e0888d5296ba783f1b58e4f3b

**_Source Branch_** (master/stable) **:**
Master

**_Additional Information_** (Steps to reproduce/Expected behavior) **:**
Mission 6-1, after accepting from gate guard, Halver is not setting the MissionStatus variable to 1, which means the mission will not progress.

The current code links are:
`
elseif (currentMission == LEAUTE_S_LAST_WISHES and MissionStatus == 0) then
            player:startEvent(25);
`

Then event 25 is:
`    elseif (csid == 25) then
        player:setVar("MissionStatus",1);`

The odd part is, I get the proper dialog, but the MissionStatus variable is not being set. At first I thought it was because it was looking for the MissionStatus variable to be set to 0, but...I am not sure.

**Comment by TeoTwawki** [comment]
I see this happen to people randomly if they have multiple characters connected from the same external IP.

Keyitems too. Their get whatever cs/dialog they were supposed to and there are no errors in log, but they are missing a variable change or a keyitem they were supposed to receive that the onEventFinish should have applied.

**Comment by helixhamin** [comment]
Yeah, I had to just gm in the variable, and I was x3 boxing, so...
Still, what happens if you have two people playing from the same internal network? Would this still happen? I mean, dual/trio boxing is one thing, but what about those that may just be on the same network but different computers?

**Comment by TeoTwawki** [comment]
Yes, still happens. My nephew plays from same connection as me, we get this all the time.

**Comment by abriasffxi** [comment]
I have had it happen about 15 times on various missions in the last 3 weeks.  Theres 6 of us that went from rank 1-6 so maybe 60 cutscenes each?   Nobody is dual boxing or on the same external IP.

**Comment by TeoTwawki** [comment]
I will do some digging either late tonight or tomorrow morning, gotta be some optional bit or some sequence breaking to cause it since most players have gotten by this without getting that error.

**Comment by TeoTwawki** [comment]
> various missions

gotta be more specific for me to look into though...

**Comment by abriasffxi** [comment]
Sorry was away for the weekend.  One was Sandy 3-1, Infiltrate Davoi.  Zoning in to Davoi, the cutscene showed for two people, but the NPC near the bridge wouldn't talk to them.  I had to @delmission them, and start over.  On the second time, one of them had the same issue in the same spot, but he logged out all the way to desktop, logged back in, and the NPC talked to him and everything worked from there.

It also happened a few times during 3-3 when we tried to talk to the King someone had a problem, and when returning to the ambassador in Re'Lude a different person didn't get the cutscene.  We fixed both by logging out to desktop since we kinda figured out that solved it the first time.

**Comment by TeoTwawki** [comment]
Theres about zero chance that there was a code bug if it suddenly worked without us changing the script..
